### ISSUE TITLE
Bug with nested templating strings

### ISSUE THREAD

**Issue by mindreframer** [zachallaun/mneme#29]
Hey, it's me again! I seem to push Mneme rather hard. Another little bug. 

```elixir
test "bug" do
    res =
      {:ok,
       %{
         database: {:literal, "\"my_app_test\#{System.get_env(\"MIX_TEST_PARTITION\")}\""}
       }}

    assert res ==
             {:ok,
              %{database: {:literal, "\"my_app_test\#{System.get_env(\"MIX_TEST_PARTITION\")}\""}}}

    ### THIS fails
    auto_assert(res)
end
```

Message: 

```
 ** (Mneme.InternalError) Mneme encountered an internal error. This is likely a bug in Mneme.

     Please consider reporting this error at https://github.com/zachallaun/mneme/issues. Thanks!

     ** (SyntaxError) nofile:18:67: unexpected token: "\" (column 67, code point U+005C)
         |
      18 |       {:ok, %{database: {:literal, "\"my_app_test#{System.get_env(\"MIX_TEST_PARTITION\")}\""}}} <-
         |                                                                   ^
         (sourceror 0.12.2) lib/sourceror.ex💯 Sourceror.parse_string!/1
         (rewrite 0.6.3) lib/rewrite/source.ex:649: Rewrite.Source.get_modules/1
         (rewrite 0.6.3) lib/rewrite/source.ex:338: Rewrite.Source.update_modules/2
         (rewrite 0.6.3) lib/rewrite/source.ex:319: Rewrite.Source.update/3
         (mneme 0.2.5) lib/mneme/patcher.ex:106: Mneme.Patcher.patch_assertion/3
         (mneme 0.2.5) lib/mneme/patcher.ex:44: Mneme.Patcher.patch!/4
         (mneme 0.2.5) lib/mneme/server.ex:149: Mneme.Server.do_patch_assertion/2
         (mneme 0.2.5) lib/mneme/server.ex:139: Mneme.Server.handle_continue/2
         (stdlib 4.2) gen_server.erl:1123: :gen_server.try_dispatch/4
         (stdlib 4.2) gen_server.erl:865: :gen_server.loop/7
         (stdlib 4.2) proc_lib.erl:240: :proc_lib.init_p_do_apply/3
```

Reproducible repo: 
- https://github.com/mindreframer/mneme_bug_dbg/commit/7c1cada6ad39272d4180151d2a4f0a558ef5fee9

**Comment by zachallaun** [comment]
@mindreframer Went ahead and released v0.2.7 with the fix.

**Comment by mindreframer** [comment]
@zachallaun Awesome! I don't know, how you do this, but to say it again - this is crazy good bug resolution time. The highlighting bug does not bother me, I'm using mostly auto-accept during development. 

Thanks!
